crates/subspace-service/src/pool.rs

```diff
@@ -121,16 +127,48 @@ where
         match self.client.runtime_api().extract_fraud_proof(at, &uxt) {
             Ok(maybe_fraud_proof) => {
                 if let Some(fraud_proof) = maybe_fraud_proof {
-                    if let Err(err) = self.verifier.verify_fraud_proof(&fraud_proof) {
-                        tracing::debug!(target: "txpool", error = ?err, "Invalid fraud proof");
-                        return async move {
-                            Err(TxPoolError::InvalidTransaction(
-                                pallet_executor::InvalidTransactionCode::FraudProof.into(),
-                            )
-                            .into())
+                    let inner = self.inner.clone();
+                    let spawner = self.spawner.clone();
+                    let fraud_proof_verifier = self.verifier.clone();
+                    let at = *at;
+
+                    return async move {
```

Comment:
Just wondering, why not `return spawner.spawn_blocking` then, so we have less wrappers and don't need to clone spawner?

Reply:
Return type doesn't worm like that, sorry